device/core/src/module_client.ts

```diff
@@ -222,4 +223,90 @@ export class ModuleClient extends EventEmitter {
   static fromAuthenticationProvider(authenticationProvider: AuthenticationProvider, transportCtor: any): ModuleClient {
     return InternalClient.fromAuthenticationProvider(authenticationProvider, transportCtor, ModuleClient) as ModuleClient;
   }
+
+  static validateEnvironment(): void {
+    // Codes_SRS_NODE_MODULE_CLIENT_13_029: [ If environment variables EdgeHubConnectionString and IotHubConnectionString do not exist then the following environment variables must be defined: IOTEDGE_WORKLOADURI, IOTEDGE_DEVICEID, IOTEDGE_MODULEID, IOTEDGE_IOTHUBHOSTNAME, IOTEDGE_AUTHSCHEME and IOTEDGE_MODULEGENERATIONID. ]
+
+    const keys = [
+      'IOTEDGE_WORKLOADURI',
+      'IOTEDGE_DEVICEID',
+      'IOTEDGE_MODULEID',
+      'IOTEDGE_IOTHUBHOSTNAME',
+      'IOTEDGE_AUTHSCHEME',
+      'IOTEDGE_MODULEGENERATIONID'
+    ];
+
+    keys.forEach((key) => {
+      if (!process.env[key]) {
+        throw new errors.PreconditionFailedError(
+          `Envrionment variable ${key} was not provided.`
+        );
+      }
+    });
+
+    // Codes_SRS_NODE_MODULE_CLIENT_13_030: [ The value for the environment variable IOTEDGE_AUTHSCHEME must be SasToken. ]
+
+    // we only support sas token auth scheme at this time
+    if (process.env.IOTEDGE_AUTHSCHEME !== 'SasToken') {
+      throw new errors.PreconditionFailedError(
+        `Authentication scheme ${
+          process.env.IOTEDGE_AUTHSCHEME
+        } is not a supported scheme.`
+      );
+    }
```

Comment:
same thing about the error: also, I don't see why we want the user to add this env variable if there's only one possible value. I'd advocate to set SasToken to be the default, and ask for the env variable only when we actually have the choice of multiple values. unnecessary burden otherwise.

Reply:
I want to leave this in since we'll also support x509 auth scheme shortly after GA. I'd rather have this check be in here rather than forgetting to validate this later when the new scheme is added.